packages/devtools_app/lib/src/service/service_manager.dart

```diff
@@ -539,6 +539,23 @@ class ServiceConnectionManager {
     await whenValueNonNull(isolateManager.mainIsolate);
     return libraryUriAvailableNow(uri);
   }
+
+  Future<String?> rootLibraryForSelectedIsolate() async {
+    if (!connectedState.value.connected) return null;
+
+    final selectedIsolateRef = isolateManager.mainIsolate.value?.id;
```

Comment:
I don't think this can be null if `connectedState.value.connected == true`.

Reply:
Ack. I traced the code back and technically `_computeMainIsolate` could return null in the isolate managers initialization. I'd rather fail gracefully here than blow up for some corner case.